_posts/2023/08/2023-08-14-memory.md

```diff
@@ -0,0 +1,59 @@
+---
+layout: post
+date: 2023-08-04
+title: "The MEMORY Object Update"
+author: graur
+---
+
+In the recently released version [0.31.0](https://github.com/objectionary/eo/releases/tag/0.31.0), we have changed the writing 
+mechanism of the `memory` object. Previously, we could store and write any object to the `memory`. However, with the update, 
+you can no longer write an object with a different type to the `memory`. Only objects of type `int`, `float`, `string`, 
+and `bool` are allowed.
+
+<!--more-->
+
+We believe that using the `memory` object as a storage for any object is incorrect. Therefore, we have modified the logic of 
+how objects are saved in the `memory` object. Let's consider an example before this update:
+
+```
+[] > animal
+  memory 0 > voice
+
+[] > dog
+  animal.voice.write "Woof" > bark  
+```
+
+In this example, we have two objects: `animal` and `dog`. The `animal` object has a `voice` object initialized as `memory` with zero. The `dog` object only has a `bark` object. We then write the string "Woof" to the `voice` object of the animal. It looks strange to write an `int` object first and then try to write a string to it.
```

Comment:
@Graur "we then write the string" seems a bit wrong, "then" can't be right after subject "we"